copy/copy.go

```diff
@@ -157,6 +158,19 @@ func Image(policyContext *signature.PolicyContext, destRef, srcRef types.ImageRe
 		}
 	}()
 
+	if !dest.AcceptsAnyOSSpecificImage() {
+		c, err := src.OCIConfig()
+		if err != nil {
+			return err
+		}
+		osErr := fmt.Errorf("image operating system %q cannot be used on %q", c.OS, runtime.GOOS)
+		if runtime.GOOS == "windows" && c.OS == "linux" {
+			return osErr
+		} else if runtime.GOOS != "windows" && c.OS == "windows" {
+			return osErr
+		}
+	}
```

Comment:
Yeah, please move this into a separate helper function to make `copy.Image` smaller.